libutils/file_lib.c

```diff
@@ -993,9 +993,24 @@ FILE *safe_fopen_create_perms(
         return NULL;
     }
 
+    char fdopen_mode_str[4] = {0};
+
     int flags = 0;
     for (int c = 0; mode[c]; c++)
     {
+        /* Ignore mode letters not supported in fdopen() for the mode string
+         * that we will use in fdopen(). Mode letters like 'x' will for
+         * example cause fdopen() to return error on Windows. */
+        if (strchr("rwa+b", mode[c]) != NULL)
+        {
+            size_t len = strlen(fdopen_mode_str);
```

Comment:
we can simply use an `idx++` and check `idx`

Reply:
Yup, nice performance boost there :stuck_out_tongue_winking_eye: :beers: